### DISCUSSION TITLE
FastDDSGen detects IDL file changes

### DISCUSSION THREAD

**Discussion by anwaymukherjee** [eProsima/Fast-DDS-Gen#314] [Ideas]
Hi,

Does fastdds generator have a way to detect any IDL file changes and then generates the code accordingly ?
Like, in cmake we have timestamp-based detection to tag a file as changed and then build. I am trying to find out if the generator can automatically skip code generation if the content of IDL has not changed.

Thanks

**Comment by JLBuenoLopez** [comment]
I am afraid that Fast DDS-Gen does not support the feature you describe. You may use the `-replace` option to overwrite the already generated files. Without enabling this option, Fast DDS-Gen would not generate the code again, even if the IDL files have changed.

I am moving this ticket to the proper discussion forum.
